include/forcing/NullForcingProvider.hpp

```diff
@@ -0,0 +1,271 @@
+            return true;
+        }
+        if (name == CSDMS_STD_NAME_SOLAR_LONGWAVE) {
+            return true;
+        }
+        if (name == CSDMS_STD_NAME_LIQUID_EQ_PRECIP_RATE) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Get whether a property's per-time-step values are each an aggregate sum over the entire time step.
+     *
+     * Certain properties, like rain fall, are aggregated sums over an entire time step.  Others, such as pressure,
+     * are not such sums and instead something else like an instantaneous reading or an average value.
+     *
+     * It may be the case that forcing data is needed for some discretization different than the forcing time step.
+     * This aspect must be known in such cases to perform the appropriate value interpolation.
+     *
+     * @param name The name of the forcing property for which the current value is desired.
+     * @return Whether the property's value is an aggregate sum.
+     */
+    //TODO this one used in Bmi_Module_Formulation.hpp and Bmi_Multi_Formulation.hpp
+    inline bool is_property_sum_over_time_step(const std::string& name) override {
+        return is_param_sum_over_time_step(name);
+    }
+
+    const std::vector<std::string> &get_available_variable_names() override {
+        return available_forcings;
+    }
+
+    private:
+
+    /**
+     * @brief Checks forcing vector index bounds and adjusts index if out of vector bounds
+     * /// \todo: Bounds checking is based on precipitation vector. Consider potential for vectors of different sizes and indices.
+     */
+    inline void check_forcing_vector_index_bounds()
+    {
+        //Check if forcing index is less than zero and if so, set to zero.
+        if (forcing_vector_index < 0)
+        {
+            forcing_vector_index = 0;
+            /// \todo: Return appropriate warning
+            std::cout << "WARNING: Forcing vector index is less than zero. Therefore, setting index to zero." << std::endl;
+        }
+
+        //Check if forcing index is greater than or equal to the size of the size of the time vector and if so, set to zero.
+        else if (forcing_vector_index >= time_epoch_vector.size())
+        {
+            forcing_vector_index = time_epoch_vector.size() - 1;
+            /// \todo: Return appropriate warning
+            std::cout << "WARNING: Reached beyond the size of the forcing vector. Therefore, setting index to last value of the vector." << std::endl;
+        }
+
+        return;
+    }
+
+    /**
+     * Get the current value of a forcing param identified by its name.
+     *
+     * @param name The name of the forcing param for which the current value is desired.
+     * @param index The index of the desired forcing time step from which to obtain the value.
+     * @return The particular param's value at the given forcing time step.
+     */
+    inline double get_value_for_param_name(const std::string& name, int index) {
```

Comment:
This looks like it was only used internally in CsvPerFeatureForcingProvider and isn't used in this class, so should be removed.

Reply:
Will test it.